superset-frontend/src/explore/components/controls/FilterBoxItemControl/FilterBoxItemControl.test.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import userEvent from '@testing-library/user-event';
+import FilterBoxItemControl from '.';
+
+const datasource = {
+  columns: [],
+  metrics: [],
+};
+
+let props = {
```

Comment:
Do you really need `beforeEach` for this case? It seems that the properties are not being modified by test cases so we could use a constant like `datasource`.

Reply:
Since many of these functions have no tests and we have nothing guaranteeing immutability, I prefer to redefine the props for all executions. It is the only way to ensure that it is in fact the prop that is being used.

As we do not have many backers, I prefer to follow a more conservative path and ensure that I am in fact testing what I think I am testing.

Reply:
I think we have some options to improve the readability, especially when the default props are big trees of values:
- For the tests where we have plain object props the spread operator `{...props}` is already guaranteeing immutability. In this case, we can define the default props as a `const` and override them in each test.
- For more complex structures we could use Lodash `cloneDeep` and follow the same pattern.
- If we want to use `beforeEach`, and assert that the default props have the correct type, we can do something like:

```
const newDefaultProps = () => ({
  name: 'John',
  lastname: 'Doe',
  age: 10,
});

let defaultProps: ReturnType<typeof newDefaultProps>;

beforeEach(() => {
  defaultProps = newDefaultProps();
});
```

Reply:
> * For the tests where we have plain object props the spread operator `{...props}` is already guaranteeing immutability. In this case, we can define the default props as a `const` and override them in each test.

This is very specific. It would only work for simpler objects (which is not the case most of the time)



> * For more complex structures we could use Lodash `cloneDeep` and follow the same pattern.

It is a good option when we do not have many functions. But I would like to avoid using more things than necessary in the tests ...

> * If we want to use `beforeEach`, and assert that the default props have the correct type, we can do something like:

Use factories! I loved this option, we didn't add external dependencies and we always have a new object. We can even remove the beforeEach in some cases. In the future we can add tests to check the immutability of the parameters. great idea!